orttraining/orttraining/training_ops/cuda/optimizer/adam/adam.h

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <vector>
+
+#include "core/common/common.h"
+#include "core/providers/cuda/cuda_kernel.h"
+
+namespace onnxruntime {
+namespace cuda {
+
+class Adam final : public CudaKernel {
+ public:
+  Adam(const OpKernelInfo& info) : CudaKernel(info) {
+    info.GetAttrOrDefault("alpha", &alpha_, 0.9f);
+    info.GetAttrOrDefault("beta", &beta_, 0.999f);
+    info.GetAttrOrDefault("epsilon", &epsilon_, 1e-8f);
+
+    info.GetAttrOrDefault("weight_decay", &weight_decay_, 0.f);
+    info.GetAttrOrDefault("adam_mode", &adam_mode_, static_cast<int64_t>(0));
+    info.GetAttrOrDefault("correct_bias", &correct_bias_, static_cast<int64_t>(1));
+
+    ORT_ENFORCE(CheckIntAttributeValid(adam_mode_, {0, 1}),
```

Comment:
given there are only 2 valid values, it might be easier to just validate that the given value is either 0 or 1.